src/linker/utilities/general_utils.py

```diff
@@ -88,7 +87,7 @@ def _add_logging_sink(
         )
 
 
-def create_results_directory(output_dir: Optional[str], timestamp: bool) -> Path:
+def create_results_directory(output_dir: Union[str, None], timestamp: bool) -> Path:
```

Comment:
Optional[str] means str or None IIRC.

Reply:
It does, and I spent a few minutes googling when to use one vs the other. I prefer it here even though it's not as modern b/c the variable is ultimately managed by the click package and it feels to me that it's not really "optional". 

But I really don't care much either way.